subprojects/logging/src/main/java/org/gradle/internal/logging/console/WorkInProgressRenderer.java

```diff
@@ -208,4 +227,48 @@ void renderNow() {
             label.setText(labelFormatter.format(operation));
         }
     }
+
+    private static class ParentRegistry {
+        private final Map<OperationIdentifier, Set<OperationIdentifier>> parentIdToChildrenIds = new HashMap<OperationIdentifier, Set<OperationIdentifier>>();
```

Comment:
⭕️  Is there an opportunity to add functionality to `ProgressOperations` instead of maintaining the progress operation hierarchy 2 ways?